themes/powerline/powerline.base.bash

```diff
@@ -1,282 +1,280 @@
-    echo "${AWS_PROFILE_CHAR}${AWS_PROFILE}|${AWS_PROFILE_PROMPT_COLOR}"
-  fi
+function __powerline_aws_profile_prompt() {
+	if [[ -n "${AWS_PROFILE}" ]]; then
+		echo "${AWS_PROFILE_CHAR}${AWS_PROFILE}|${AWS_PROFILE_PROMPT_COLOR}"
+	fi
 }
 
-function __powerline_shlvl_prompt {
-  if [[ "${SHLVL}" -gt 1 ]]; then
-    local prompt="${SHLVL_THEME_PROMPT_CHAR}"
-    local level=$(( ${SHLVL} - 1))
-    echo "${prompt}${level}|${SHLVL_THEME_PROMPT_COLOR}"
-  fi
+function __powerline_shlvl_prompt() {
+	if [[ "${SHLVL}" -gt 1 ]]; then
+		local prompt="${SHLVL_THEME_PROMPT_CHAR}"
+		local level=$((SHLVL - 1))
+		echo "${prompt}${level}|${SHLVL_THEME_PROMPT_COLOR}"
+	fi
 }
 
-function __powerline_dirstack_prompt {
-  if [[ "${#DIRSTACK[@]}" -gt 1 ]]; then
-    local depth=$(( ${#DIRSTACK[@]} - 1 ))
-    local prompt="${DIRSTACK_THEME_PROMPT_CHAR}"
-    if [[ "${depth}" -ge 2 ]]; then
-      prompt+="${depth}"
-    fi
-    echo "${prompt}|${DIRSTACK_THEME_PROMPT_COLOR}"
-  fi
+function __powerline_dirstack_prompt() {
+	if [[ "${#DIRSTACK[@]}" -gt 1 ]]; then
+		local depth=$((${#DIRSTACK[@]} - 1))
+		local prompt="${DIRSTACK_THEME_PROMPT_CHAR}"
+		if [[ "${depth}" -ge 2 ]]; then
+			prompt+="${depth}"
+		fi
+		echo "${prompt}|${DIRSTACK_THEME_PROMPT_COLOR}"
+	fi
 }
 
-function __powerline_history_number_prompt {
-  echo "${HISTORY_NUMBER_THEME_PROMPT_CHAR}\!|${HISTORY_NUMBER_THEME_PROMPT_COLOR}"
+function __powerline_history_number_prompt() {
+	echo "${HISTORY_NUMBER_THEME_PROMPT_CHAR}\!|${HISTORY_NUMBER_THEME_PROMPT_COLOR}"
 }
 
-function __powerline_command_number_prompt {
-  echo "${COMMAND_NUMBER_THEME_PROMPT_CHAR}\#|${COMMAND_NUMBER_THEME_PROMPT_COLOR}"
+function __powerline_command_number_prompt() {
+	echo "${COMMAND_NUMBER_THEME_PROMPT_CHAR}\#|${COMMAND_NUMBER_THEME_PROMPT_COLOR}"
 }
 
-function __powerline_left_segment {
-  local OLD_IFS="${IFS}"; IFS="|"
-  local params=( $1 )
-  IFS="${OLD_IFS}"
-  local pad_before_segment=" "
-
-  if [[ "${SEGMENTS_AT_LEFT}" -eq 0 ]]; then
-    if [[ "${POWERLINE_COMPACT_BEFORE_FIRST_SEGMENT}" -ne 0 ]]; then
-      pad_before_segment=""
-    fi
-  else
-    if [[ "${POWERLINE_COMPACT_AFTER_SEPARATOR}" -ne 0 ]]; then
-      pad_before_segment=""
-    fi
-    # Since the previous segment wasn't the last segment, add padding, if needed
-    #
-    if [[ "${POWERLINE_COMPACT_BEFORE_SEPARATOR}" -eq 0 ]]; then
-      LEFT_PROMPT+="$(set_color - ${LAST_SEGMENT_COLOR}) ${normal}"
-    fi
-    if [[ "${LAST_SEGMENT_COLOR}" -eq "${params[1]}" ]]; then
-      LEFT_PROMPT+="$(set_color - ${LAST_SEGMENT_COLOR})${POWERLINE_LEFT_SEPARATOR_SOFT}${normal}"
-    else
-      LEFT_PROMPT+="$(set_color ${LAST_SEGMENT_COLOR} ${params[1]})${POWERLINE_LEFT_SEPARATOR}${normal}"
-    fi
-  fi
-
-  LEFT_PROMPT+="$(set_color - ${params[1]})${pad_before_segment}${params[0]}${normal}"
-  LAST_SEGMENT_COLOR=${params[1]}
-  (( SEGMENTS_AT_LEFT += 1 ))
+function __powerline_left_segment() {
+	local params
+	IFS="|" read -ra params <<< "${1}"
```

Comment:
This is the most complicated change in the PR.
TIL about the `read` function.
Testing locally confirmed that, if you create the `local` variable first, then the read operation will populate the existing variable (vs creating a new, global variable)